apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/EndpointJsonResponseComparatorV2E2E.java

```diff
@@ -0,0 +1,865 @@
+package gov.cms.bfd.server.war.r4.providers;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import ca.uhn.fhir.model.primitive.IdDt;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.google.common.collect.Lists;
+import com.newrelic.relocated.JsonArray;
+import gov.cms.bfd.model.rif.entities.Beneficiary;
+import gov.cms.bfd.model.rif.samples.StaticRifResourceGroup;
+import gov.cms.bfd.server.war.EndpointJsonComparatorBase;
+import gov.cms.bfd.server.war.ServerTestUtils;
+import gov.cms.bfd.server.war.commons.ClaimType;
+import gov.cms.bfd.server.war.commons.MedicareSegment;
+import gov.cms.bfd.server.war.commons.TransformerConstants;
+import io.restassured.http.Header;
+import io.restassured.http.Headers;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * This set of tests compare the application's current responses to a set of previously-recorded
+ * responses. This achieves several goals:
+ *
+ * <ul>
+ *   <li>It helps us to ensure that we're not accidentally changing the application's responses
+ *   <li>It helps us to maintain backwards compatibility.
+ *   <li>As any changes in an operation's output will have to include a change to the recorded
+ *       response, it makes it much easier to tell what our PRs are actually doing.
+ * </ul>
+ *
+ * <p>There SHALL be a 1:1 relationship between test cases here and the application's operations;
+ * every supported operation should have a test case.
+ *
+ * <p>Note that our responses include timestamps and have other differences from request to request
+ * (e.g. element ordering). Each test case must ignore or otherwise work around such differences so
+ * that tests work reliably.
+ *
+ * <p>To re-generate the recorded responses, build the application with -DgenerateTestData=true
+ * which will run the test that creates the endpoint responses.
+ */
+public class EndpointJsonResponseComparatorV2E2E extends EndpointJsonComparatorBase {
```

Comment:
This and the other Comparator test were both renamed and moved, so GitHub is understandably confused. The classes were not completely rewritten, but were significantly changed in the following locations:
- Many utility methods moved into a base class or the ServerTestUtils
- Test method bodies (metadata, patientRead, etc) were changed to use RestAssured; compare any one of them to the "deleted" class version to see the improvement
- The "sort" methods were mostly untouched, except for minor touchups (using modern loops/streams, fixing useless initializers)